react/ShippingCalculator.tsx

```diff
@@ -48,20 +54,24 @@ const ShippingCalculator: React.VFC = () => {
 
   return (
     <>
-      <ShippingHeader
-        onEditAddress={() => {
-          setEditingAddress(true)
-          setAddress(null)
-        }}
-      />
-
       {!editingAddress ? (
-        <ShippingOptionList
-          deliveryOptions={deliveryOptions}
-          onDeliveryOptionSelected={selectDeliveryOption}
-        />
+        <div>
+          <ShippingHeader
+            onEditAddress={() => {
+              setEditingAddress(true)
+              setAddress(null)
+            }}
+          />
+
+          <ShippingOptionList
+            deliveryOptions={deliveryOptions}
+            pickupOptions={pickupOptions}
+            onDeliveryOptionSelected={selectDeliveryOption}
+            onPickupOptionSelected={selectPickupOption}
+          />
+        </div>
       ) : (
-        <LocationInput onSuccess={handleAddressSuccess} />
+        <NewAddressForm onAddressCreated={handleAddressSuccess} />
```

Comment:
shouldn't we add the `ShippingHeader` component here as well?

Reply:
On  `checkout-shipping` it's used only at the shipping options listing. And also, if I add it to the input phase, it gets weird.

![image](https://user-images.githubusercontent.com/12574426/100794419-4f4aa800-33fc-11eb-952f-f4c11123bf5e.png)

Reply:
I think I've solved this before 🤔

the header should be displaying the "Por favor informe seu endereço" message in this stage

Reply:
I'm gonna check this and open a PR with a possible fix, ok?

Reply:
But I think for now we could merge it like this.

Reply:
I think this is a design regression that we should solve before releasing 😅

there is also some unnecessary space between the location input and the device coordinators that I don't know where is coming from (it's an empty `<div>`). @wdsrocha do you know anything about it? maybe it's the container for location search, but I'm not sure

Reply:
> there is also some unnecessary space between the location input and the device coordinators that I don't know where is coming from (it's an empty `<div>`).

oh, I know what it is! it's the container for location country... I think to solve it we can add a `className` prop to the component, so the margin will only take effect when the location country is actually rendered

Reply:
For now, I fixed it with [this commit](https://github.com/vtex-apps/checkout-shipping/pull/18/commits/98c1eefbaf2178045e3b060973ffe87eb0db855e) on the checkout-shipping PR. Instead of just checking for the address, I'm now checking if the postal code exists. But I think we should update the function that determines the `invalidFields` on the `vtex.address-context` app in the future.

Reply:
> > there is also some unnecessary space between the location input and the device coordinators that I don't know where is coming from (it's an empty `<div>`).
> 
> oh, I know what it is! it's the container for location country... I think to solve it we can add a `className` prop to the component, so the margin will only take effect when the location country is actually rendered

I'm not following you 🤔 . Could you explain it again like if you were talking to a 10 years old kid? 😆

Reply:
> I'm not following you 🤔 . Could you explain it again like if you were talking to a 10 years old kid? 😆

sure 😆

we have the following empty div which is the container for the `LocationCountry` component

<img width="431" alt="image" src="https://user-images.githubusercontent.com/10223856/101018215-f0934480-3549-11eb-9c86-1829551b9bd7.png">

but for this scenario where we don't have multiple countries in the deliveries, the component isn't rendered:

https://github.com/vtex-apps/place-components/blob/0632b51976e0cdd555ba7bdf7f689469f77410ca/react/LocationCountry.tsx#L101-L104

which makes us render the `<div>` containing the component, even though it didn't actually render anything, adding this unneeded margin

Reply:
Noice! Since I already merged the `checkout-shipping` PR, I'll open another one with this change. But could you this one again, please?